src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.CoreRT.cs

```diff
@@ -0,0 +1,476 @@
+        }
+
+        internal unsafe struct ManagedObjectWrapper
+        {
+            public volatile IntPtr Target; // This is GC Handle
+            public long RefCount;
+
+            public int UserDefinedCount;
+            public ComInterfaceEntry* UserDefined;
+            internal InternalComInterfaceDispatch* Dispatches;
+
+            internal volatile CreateComInterfaceFlagsEx Flags;
+            const ulong ComRefCountMask = 0x000000007fffffffUL;
+            static uint GetComCount(ulong c)
+            {
+                return (uint)(c & ComRefCountMask);
+            }
+
+            public uint AddRef()
+            {
+                return GetComCount((ulong)Interlocked.Increment(ref RefCount));
+            }
+
+            public uint Release()
+            {
+                if (GetComCount((ulong)RefCount) == 0)
+                {
+                    Debug.Fail("Over release of MOW - COM");
+                    return unchecked((uint)-1);
+                }
+
+                return GetComCount((ulong)Interlocked.Decrement(ref RefCount));
+            }
+
+            public unsafe int QueryInterface(in Guid riid, out IntPtr ppvObject)
+            {
+                ppvObject = AsRuntimeDefined(in riid);
+                if (ppvObject == IntPtr.Zero)
+                {
+                    ppvObject = AsUserDefined(in riid);
+                    if (ppvObject == IntPtr.Zero)
+                        return HResults.COR_E_INVALIDCAST;
+                }
+
+                AddRef();
+                return HResults.S_OK;
+            }
+
+            public IntPtr As(in Guid riid)
+            {
+                // Find target interface and return dispatcher or null if not found.
+                IntPtr typeMaybe = AsRuntimeDefined(in riid);
+                if (typeMaybe == IntPtr.Zero)
+                    typeMaybe = AsUserDefined(in riid);
+
+                return typeMaybe;
+            }
+
+            unsafe IntPtr AsRuntimeDefined(in Guid riid)
+            {
+                if ((Flags & CreateComInterfaceFlagsEx.CallerDefinedIUnknown) == CreateComInterfaceFlagsEx.None)
+                {
+                    if (riid == IID_IUnknown)
+                    {
+                        return (IntPtr)(Dispatches + UserDefinedCount);
+                    }
+                }
+
+                return IntPtr.Zero;
+            }
+
+            unsafe IntPtr AsUserDefined(in Guid riid)
+            {
+                for (int i = 0; i < UserDefinedCount; ++i)
+                {
+                    if (UserDefined[i].IID == riid)
+                    {
+                        return (IntPtr)(Dispatches + i);
+                    }
+                }
+
+                return IntPtr.Zero;
+            }
+        }
+
+        internal unsafe class ManagedObjectWrapperHolder
+        {
+            private ManagedObjectWrapper* wrapper;
```

Comment:
```suggestion
            private ManagedObjectWrapper* _wrapper;
```
to follow the coding conventions.

Reply:
Also, do not use `this.` unless absolutely required.